agent/command/s3_util.go

```diff
@@ -88,5 +88,22 @@ func getAssumedRoleARN(conf *internal.TaskConfig, sessionToken string) string {
 		return ""
 	}
 
-	return conf.AssumeRoleRoles[sessionToken]
+	return conf.AssumeRoleRoles[sessionToken].RoleARN
+}
+
+// getAssumedRoleExpiration checks if the provided session token
+// is associated with an assumed role. If it is, it returns
+// the expiration time of the role.
+func getAssumedRoleExpiration(conf *internal.TaskConfig, sessionToken string) (time.Time, bool, error) {
+	if sessionToken == "" {
+		return time.Time{}, false, nil
+	}
+	expiration := conf.AssumeRoleRoles[sessionToken].Expiration
+	if expiration == "" {
+		return time.Time{}, false, nil
+	}
+
+	exp, err := time.Parse(time.RFC3339, expiration)
+
+	return exp, true, errors.Wrap(err, "parsing expiration time")
```

Comment:
nit: may help debugging to include the expiration time string, I'm not sure if `time.Parse` includes it in the error.